samples/nrf9160/lwm2m_client/src/main.c

```diff
@@ -57,19 +57,27 @@ LOG_MODULE_REGISTER(app_lwm2m_client, CONFIG_APP_LOG_LEVEL);
 #define LWM2M_SECURITY_CERTIFICATE 2
 #define LWM2M_SECURITY_NO_SEC 3
 
+/* Client State Machine states*/
+enum lwm2m_client_state {
+	START,		/* Start Connection to server*/
+	CONNECTING,	/* Client is connecting to server */
+	BOOTSTRAP,	/* Client is doing bootstrap */
+	CONNECTED,	/* Client connection establisment to server */
+	LTE_OFFLINE,	/* LTE offline and LwM2M engine suspended */
+	NETWORK_ERROR	/* Client networ error handling. Client stop and modem reset */
+};
```

Comment:
There is confusion about what "client" is, so lets call the Zephyr side library as "engine" and application as a "client"

```suggestion
/* Client states*/
enum lwm2m_client_state {
	START,		/* Start Connection to a server*/
	CONNECTING,	/* LwM2M engine is connecting  */
	BOOTSTRAP,	/* LwM2M enginet is doing a bootstrap */
	CONNECTED,	/* LwM2M Client connection establisment to server */
	LTE_OFFLINE,	/* LTE offline and LwM2M engine should be suspended */
	NETWORK_ERROR	/* Client networ error handling. Client stop and modem reset */
};
```

Reply:
Ah OK let me fix this.